tests/robustness/failpoint/failpoint.go

```diff
@@ -570,3 +574,133 @@ func (f killAndGofailSleep) Available(config e2e.EtcdProcessClusterConfig, membe
 	}
 	return memberFailpoints.Available(f.failpoint)
 }
+
+type memberReplace struct{}
+
+func (f memberReplace) Inject(ctx context.Context, t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster) error {
+	memberId := rand.Int() % len(clus.Procs)
+	member := clus.Procs[memberId]
+	var endpoints []string
+	for i := 1; i < len(clus.Procs); i++ {
+		endpoints = append(endpoints, clus.Procs[(memberId+i)%len(clus.Procs)].EndpointsGRPC()...)
+	}
+	cc, err := clientv3.New(clientv3.Config{
+		Endpoints:            endpoints,
+		Logger:               zap.NewNop(),
+		DialKeepAliveTime:    50 * time.Second,
+		DialKeepAliveTimeout: 100 * time.Millisecond,
+	})
+	if err != nil {
+		return err
+	}
+	defer cc.Close()
+	memberID, found, err := getID(ctx, cc, member.Config().Name)
+	if err != nil {
+		return err
+	}
+	if !found {
+		t.Fatal("Member not found")
+	}
+	// Need to wait health interval for cluster to accept member changes
+	time.Sleep(etcdserver.HealthInterval)
+	lg.Info("Removing member", zap.String("member", member.Config().Name))
+	_, err = cc.MemberRemove(ctx, memberID)
+	if err != nil {
+		return err
+	}
+	_, found, err = getID(ctx, cc, member.Config().Name)
+	if err != nil {
+		return err
+	}
+	if found {
+		t.Fatal("Expected member to be removed")
+	}
+
+	for member.IsRunning() {
+		err = member.Kill()
+		if err != nil {
+			lg.Info("Sending kill signal failed", zap.Error(err))
+		}
+		err = member.Wait(ctx)
+		if err != nil && !strings.Contains(err.Error(), "unexpected exit code") {
+			lg.Info("Failed to kill the process", zap.Error(err))
+			return fmt.Errorf("failed to kill the process within %s, err: %w", triggerTimeout, err)
+		}
+	}
+	lg.Info("Removing member data", zap.String("member", member.Config().Name))
+	err = os.RemoveAll(member.Config().DataDirPath)
+	if err != nil {
+		return err
+	}
+
+	lg.Info("Adding member back", zap.String("member", member.Config().Name))
+	removedMemberPeerUrl := member.Config().PeerURL.String()
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+		reqCtx, cancel := context.WithTimeout(ctx, time.Second)
+		_, err = cc.MemberAdd(reqCtx, []string{removedMemberPeerUrl})
+		cancel()
+		if err == nil {
+			break
+		}
+	}
+	err = patchArgs(member.Config().Args, "initial-cluster-state", "existing")
+	if err != nil {
+		return err
+	}
+	lg.Info("Starting member", zap.String("member", member.Config().Name))
+	err = member.Start(ctx)
+	if err != nil {
+		return err
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+		_, found, err := getID(ctx, cc, member.Config().Name)
```

Comment:
Code changes looks good. Just have one question here: do we need to wait for the sync of snapshot or logs here?

Reply:
If you are asking, are we sure that restarted member is serving requests, then there is no 100% percent chance that member. After a fail point injection, we check if member is healthy and we give it a second, but that's not enough to guarantee that we record any requests after.

Reply:
Thanks for the comment. LGTM

Reply:
No problem, this issue has been on my mind for long time. It's a good candidate for followup.